app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -1,12 +1,52 @@
 class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
   def google_oauth2
-    @user = User.from_omniauth(request.env['omniauth.auth'])
-    if @user.persisted?
-      flash[:notice] = I18n.t 'devise.omniauth_callbacks.success', kind: 'Google'
-      sign_in_and_redirect @user, event: :authentication
+    auth = request.env["omniauth.auth"]
```

Comment:
it looks like `google_oauth2` and `trello` methods have a lot of common code. Is it possible to refactor it somehow?